week6/unique0902/src/ProductTable.jsx

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import ProductCategoryRow from './ProductCategoryRow';
+import ProductRow from './ProductRow';
+
+const ProductTable = ({
+  products,
+  filterText,
+  inStockOnly,
+  deleteProduct,
+  editProduct,
+}) => {
+  const rows = [];
+  let lastCategory = null;
+  let nowTimeKey = Date.now();
+  const productsCopy = [...products];
+  productsCopy
+    .sort((a, b) => (a.category > b.category ? 1 : -1))
+    .map((product) => {
+      if (product.name.toLowerCase().indexOf(filterText.toLowerCase()) === -1) {
+        return;
+      }
+
+      if (inStockOnly && !product.stocked) {
+        return;
+      }
+
+      if (product.category !== lastCategory) {
+        rows.push(
+          <ProductCategoryRow
+            category={product.category}
+            key={product.category}
+          />
+        );
+      }
+
+      rows.push(
+        <ProductRow
+          product={product}
+          key={nowTimeKey}
+          deleteProduct={deleteProduct}
+          editProduct={editProduct}
+        />
+      );
+      nowTimeKey++;
+      lastCategory = product.category;
+    });
```

Comment:
```suggestion
  let lastCategory = null;

  const filteredProducts = products
    .sort((a, b) => (a.category > b.category ? 1 : -1))
    .filter((product) => {
      const filterTextMatch = product.name
        .toLowerCase()
        .includes(filterText.toLowerCase());
      const inStockCheck = !inStockOnly || product.stocked;
      return filterTextMatch && inStockCheck;
    });
```

제가 맑은 정신으로 다시 코드를 보니 이 방법이 맞는거 같습니다..